### ISSUE TITLE
Confusion about Visualizing Point Cloud Reconstruction Results in Figure2

### ISSUE THREAD

**Issue by LiXinghui-666** [lulutang0608/Point-BERT#29]
One point I am confused with is the Visualizing Point Cloud Reconstruction Results of Figure2 in paper.

I want to know that how is the input masked point cloud and the final reconstructed point cloud visualization done? Because the mask operation is tokens, and the MaskTransformer output is cls-head and logits, what should I do to visualize the reconstructed point cloud of Figure 2 in paper?

Finally, I also want to know whether Point-BERT can predict the complete point cloud of the model by inputting viewpoint point cloud after pre-training.

**Comment by yuxumin** [comment]
Hi, thanks for you interest in our work.

We first mask the point cloud and obtain an incomplete point cloud. Then we use the Transformer prertrained with MPM tasks to predict the missing discrete point tokens (these tokens can be translated into vectors by the vocabulary in dVAE). Finally we can recontruct the whole point cloud by the pretrained decoder in dVAE.
Point-BERT can predict the complete point cloud by adding decoder after the Transformer. The pipeline is similar to PoinTr https://arxiv.org/abs/2108.08839

**Comment by yuxumin** [comment]
Close it since no response. Feel free to re-open it if problems still exist
